### PR TITLE
Docs fixes

### PR DESCRIPTION
**Pull request by z64**
- Add brief examples of uploading a file from a disk, as `File` seems to not be getting the point across
- Fixes some broken method link and invalid return tag in some docs
- Adds docs to some other undocumented methods, as listed by `yard stats --list-undoc` (some of which on `Session` - please check that I'm understanding them right)
- Upgrades to YARD 0.9.9, as the current version throws a bunch of old `::TRUE`, etc. constant warnings because they're deprecated
- There were two methods it seems using a `Hash` type signature inconsistently with the rest of the lib

### CODE REVIEW

File: lib/discordrb/gateway.rb [hunk 1]
```diff
@@ -100,6 +100,7 @@ def initialize(session_id)
       @invalid = false
     end
 
+    # @return [true, false] whether this session is suspended
```
[the remark this answers is not in the document]
Comment (LikeLakers2) [reply lib/discordrb/gateway.rb]: Also, this comment specifically was probably meant for `Session#suspended?`.

[the remark this answers is not in the document]
Comment (z64) [reply lib/discordrb/gateway.rb]: No, it was just a mistake - YARD doesn't actually see the other methods as requiring documentation.

**Comment by meew0** [comment]
If this is done, can you autosquash to remove the fixup commit?

**Comment by z64** [comment]
Yep, done :+1:

**Pull request merged**